server_modules/file_system.js

```diff
@@ -0,0 +1,133 @@
+var fs = require('fs');
+var Q = require('q');
+var mkdirp = require('mkdirp');
+var path = require('path')
+var utils = require('./utils');
+var args = require('./args');
+
+function getDataFilePath(file) {
+    return exports.resolve(path.dirname(process.argv[1]/*TODO change this to user the args module*/), args.data, file);

```

Comment:
1. You're right, I changed it.
2. I think it would be more SRP this way, and we already require the args module
3. That was a mistake, and I changed it.
Thank you for the review! Very good comments